pkg/controller/compute/eks.go

```diff
@@ -428,11 +419,13 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return r.delete(instance, eksClient)
 	}
 
-	// Create cluster instance
-	if instance.Status.ClusterName == "" {
+	cluster, err := eksClient.Get(meta.GetExternalName(instance))
+	switch {
+	case resource.Ignore(eks.IsErrorNotFound, err) != nil:
+		return r.fail(instance, err)
+	case eks.IsErrorNotFound(err):
 		return r.create(instance, eksClient)
```

Comment:
Switches are processed in order, so this could probably just be:

```go
switch {
case eks.IsErrorNotFound(err):
    return r.create(instance, eksClient)
case err != nil:
    return r.fail(instance, err)
}
```